src/components/Mission.js

```diff
@@ -1,22 +1,40 @@
 import PropTypes from 'prop-types';
 import { Badge, Button } from 'react-bootstrap';
+import { useDispatch } from 'react-redux';
+import { joinMission, leaveMission } from '../redux/missions/missions';
 
-const Mission = ({ name, desc }) => (
-  <>
-    <td className="col-2 fw-bold">{name}</td>
-    <td className="col-6">{desc}</td>
-    <td className="col-2">
-      <Badge bg="secondary">NOT A MEMBER</Badge>
-    </td>
-    <td className="col-2">
-      <Button variant="outline-secondary">Join Mission</Button>
-    </td>
-  </>
-);
+const Mission = ({ id, name, desc, reserved }) => {
+  const dispatch = useDispatch();
+
+  return (
+    <>
+      <td className="col-2 fw-bold">{name}</td>
+      <td className="col-6">{desc}</td>
+      <td className="col-2">
+        {!reserved && <Badge bg="secondary">NOT A MEMBER</Badge>}
+        {reserved && <Badge bg="primary">Active Member</Badge>}
+      </td>
+      <td className="col-2">
+        {!reserved && (
+          <Button variant="outline-secondary" onClick={() => dispatch(joinMission(id))}>
+            Join Mission
+          </Button>
+        )}
+        {reserved && (
+          <Button variant="outline-danger" onClick={() => dispatch(leaveMission(id))}>
+            Leave Mission
+          </Button>
```

Comment:
- [  ] OPTIONAL: What if instead of making one-line logic declarations, you prepare the content of every component that needs it, that way you separate the return statement from the logic that actually chooses the content and style(s) of the component

Reply:
Great suggestion! I applied it and it does look cleaner.